x-pack/plugins/ingest_manager/server/services/agents/checkin/state_new_actions.ts

```diff
@@ -170,7 +170,10 @@ export function agentCheckinStateNewActionsFactory() {
     }
 
     const stream$ = agentPolicy$.pipe(
-      timeout(appContextService.getConfig()?.fleet.pollingRequestTimeout || 0),
+      timeout(
+        // Set a timeout 3s before the real timeout to have a chance to respond an empty response before socket timeout
+        Math.max((appContextService.getConfig()?.fleet.pollingRequestTimeout ?? 0) - 3000, 3000)
```

Comment:
does this mean we can run into an edge case where the response can come in at _almost_ the request time out limit?  let's say 2 seconds before the limit. will this logic cause that to be ignored and treated as an empty response?

Reply:
Yes we can have an edge case here
The arbitrary 3s delay try to mitigate the risk of not sending any response (the way Kibana handle socket timeout the socket  is destroyed immediately so we cannot send any data), but it can still happen if we some huge delay in the event loop.
If the response is happening like 2 seconds before the socket timeout we will send an empty response but it will be fixed during the next checkin.

Reply:
@jen-huang If I can still clarify that :)

Reply:
> it will be fixed during the next checkin

ah got it. that sounds acceptable to me 👍